### ISSUE TITLE
Cannot Create Call - Error Code 7505 "Request Authorization Tenant Mismatch."

### ISSUE THREAD

**Issue by Mark-Bailey-SingleWire** [microsoftgraph/microsoft-graph-docs#7080]
I am attempting to create a call using Example 3. My data is hosted via an Azure Portal Office Developer account. I am sending a bearer token obtained through the process described as "Get access without a user." The two users when I POST to `https://graph.microsoft.com/v1.0/communications/calls` I get: 

```
{
  "error": {
    "code": "7505",
    "message": "Request authorization tenant mismatch.",
    "innerError": {
      "request-id": "3d972a68-4d0c-4764-bb3f-904d6ee7e20a",
      "date": "2020-02-13T17:30:24"
    }
  }
}
```
What am I doing wrong?

Best, Mark

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 776d5045-0abc-cdf9-f6c0-a2242c63ad42
* Version Independent ID: 7aa5b80a-5565-b702-20ae-8d550357980a
* Content: [Create call - Microsoft Graph v1.0](https://docs.microsoft.com/en-us/graph/api/application-post-calls?view=graph-rest-1.0&tabs=http#feedback)
* Content Source: [api-reference/v1.0/api/application-post-calls.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/master/api-reference/v1.0/api/application-post-calls.md)
* Product: **cloud-communications**
* Technology: **microsoft-graph**
* GitHub Login: @ananmishr
* Microsoft Alias: **MSGraphDocsVteam**

**Comment by Mark-Bailey-SingleWire** [comment]
Thanks, I am going to create a question on Stack Overflow with a [microsoft-graph] tag.

**Comment by Mark-Bailey-SingleWire** [comment]
In comparing the [production document example 3](https://docs.microsoft.com/en-us/graph/api/application-post-calls?view=graph-rest-1.0&tabs=http#example-3-create-a-group-call-with-service-hosted-media) and the[ beta example 3](https://docs.microsoft.com/en-us/graph/api/application-post-calls?view=graph-rest-beta&tabs=http#example-3-create-a-group-call-with-service-hosted-media), one additional property (tenantid) exists on the beta document example. When I POST with that property, the call was established. 

Please correct your documentation.

**Comment by Chameleon2525** [comment]
Thanks @gclark01, you've save me a lot of time with you comment!

**Comment by alzia** [comment]
I am getting an error creating a call (using NodeJS) - the error I get is: 

{
  error: {
    code: '9999',
    message: 'Unkown internal Server Error.',
    innerError: {
      date: '2020-08-05T22:40:19',
      'request-id': '3bc5c2eb-4339-4ec3-8a6b-3d69b6d1932f'
    }
  }
}

if I make a similar call from Postman, I get this error -

{
  "error": {
    "code": "7505",
    "message": "Request authorization tenant mismatch.",
    "innerError": {
      "date": "2020-08-06T16:11:56",
      "request-id": "deb1c86c-fed7-45b3-87ae-ac96bd6279a3"
    }
  }
}
Unsure, if Postman's POST call to create a Teams call will not work because the call needs to be made from the same tenant.

The code I am using to create a call in Node JS is as follows, please feel free to suggest if I am doing anything wrong inside the code.

**var callBody = {
          "@odata.type": "#microsoft.graph.call",
          "callbackUri": "https://bot.contoso.com/callback",
          "requestedModalities": [
            "audio"
          ],
          "chatInfo": {
            "@odata.type": "#microsoft.graph.chatInfo",
            "threadId": "19%3ameeting_Mzg3ZjhlMzgtYWM5Ny00Y2YyLWJmMTQtZjJlNWRjMTg2NzRl%40thread.v2",
            "messageId": "0"
          },
          "meetingInfo": {
            "@odata.type": "#microsoft.graph.organizerMeetingInfo",
            "allowConversationWithoutHost": true
          }
        };
        
          var headers = new Headers();
          var bearer = "Bearer " + parsedBody.access_token;
          headers.append("Authorization", bearer);
          var options = {
              method: "POST",
              headers: headers
          };
          var graphEndpointUrl = "https://graph.microsoft.com/v1.0/communications/calls";
 
          const getData = async graphEndpointUrl => {
            try {
              const response = await fetch(graphEndpointUrl, {method : 'POST', headers: headers, body:callBody});
              const json = await response.json();
              console.log(json);
            } catch (error) {
              console.log(error);
            }
          };
          getData(graphEndpointUrl);**

**Comment by gclark01** [comment]
@alzia - Looks like you are missing tenantId.  Look at my comment above that should help.

**Comment by alzia** [comment]
I added the tenant id as well, did not help..:(

**Comment by alzia** [comment]
> @alzia - Looks like you are missing tenantId. Look at my comment above that should help.

I added the tenant id as well, did not help :(....

**Comment by alzia** [comment]
This is the updated code with the tenantid per @gclark01 suggestion -
        var callBody = {
          "@odata.type": "#microsoft.graph.call",
          "callbackUri": "https://bot.contoso.com/callback",
          "requestedModalities": [
            "audio"
          ],
          "mediaConfig": {
            "@odata.type": "#microsoft.graph.serviceHostedMediaConfig",
            "preFetchMedia": [
             {
               "uri": "https://cdn.contoso.com/beep.wav",
               "resourceId": "f8971b04-b53e-418c-9222-c82ce681a582"
             },
             {
               "uri": "https://cdn.contoso.com/cool.wav",
               "resourceId": "86dc814b-c172-4428-9112-60f8ecae1edb"
             }
            ],
          },
          "chatInfo": {
            "@odata.type": "#microsoft.graph.chatInfo",
            "threadId": "19:meeting_Mzg3ZjhlMzgtYWM5Ny00Y2YyLWJmMTQtZjJlNWRjMTg2NzRl@thread.v2",
            "messageId": "0"
          },
          "meetingInfo": {
            "@odata.type": "#microsoft.graph.organizerMeetingInfo",
            "organizer": {
              "@odata.type": "#microsoft.graph.identitySet",
              "user": {
                "@odata.type": "#microsoft.graph.identity",
                "id": "f2781e5f-92a0-4d45-8ea8-d5fde0b1bff4",
                "displayName": "Mod"
              }
            },
            "allowConversationWithoutHost": true,
            "tenantId": "6ffca450-b814-484a-b909-4207a268b93b"
          },
            "tenantId": "6ffca450-b814-484a-b909-4207a268b93b"
        };
        
          var headers = new Headers();
          var bearer = "Bearer " + parsedBody.access_token;
          headers.append("Authorization", bearer);
          var options = {
              method: "POST",
              headers: headers
          };
          
          var graphEndpointUrl = "https://graph.microsoft.com/v1.0/communications/calls";
          const response = fetch(graphEndpointUrl, {method : 'POST', headers: headers, body:callBody})
          .then((res) => {
            return res.json()
       })
       .then((json) => {
         console.log(json);
         // Do something with the returned data.
       });

**Comment by gclark01** [comment]
@alzia - Just tested with this in postman and this worked.

{
  "@odata.type": "#microsoft.graph.call",
  "callbackUri": "https://bot.contoso.com/callback",
  "requestedModalities": [
    "audio"
  ],
  "mediaConfig": {
    "@odata.type": "#microsoft.graph.serviceHostedMediaConfig",
    "preFetchMedia": [
     {
       "uri": "https://cdn.contoso.com/beep.wav",
       "resourceId": "f8971b04-b53e-418c-9222-c82ce681a582"
     },
     {
       "uri": "https://cdn.contoso.com/cool.wav",
       "resourceId": "86dc814b-c172-4428-9112-60f8ecae1edb"
     }
    ],
  },
  "chatInfo": {
    "@odata.type": "#microsoft.graph.chatInfo",
    "threadId": "19:meeting_Mzk2MzdjODktNDI0ZS00Mzc3LWIyZDMtM2M3MDBhMmEwMjkz@thread.v2",
    "messageId": "0"
  },
  "meetingInfo": {
    "@odata.type": "#microsoft.graph.organizerMeetingInfo",
    "organizer": {
      "@odata.type": "#microsoft.graph.identitySet",
      "user": {
        "@odata.type": "#microsoft.graph.identity",
        "id": "9901ceb5-4766-454a-82b8-7c50988c985c",
        "displayName": "Bob",
        "tenantId": "a9aa0fb4-6235-4655-81ce-4eb467f17fc7"
      }
    },
    "allowConversationWithoutHost": true
  },
  "tenantId": "a9aa0fb4-6235-4655-81ce-4eb467f17fc7"
}

**Comment by nomaannomii** [comment]
Hi folks,

I am getting the same error even after entering tenantId. Below is my call code 

function answerCall(CommID) {
    let answergraphURL = 'https://graph.microsoft.com/beta/communications/calls/' + CommID + '/answer';
    console.log('This is the endpoint to answer: ' + answergraphURL);
	request.post(
		{
			url:answergraphURL,
			json: 
			{
				"callbackUri": process.env.ngrok + "/api/calls/hub",
				"tenantId": process.env.tenantID,
				"acceptedModalities": [ "audio" ],
				"mediaConfig": {
					"@odata.type": "#microsoft.graph.serviceHostedMediaConfig"
				}
			}
			
	}, function (err, httpResponse, body) {
		console.log('err: '+err);
		console.log('httpResponse: '+httpResponse);
		console.log('body: '+JSON.stringify(body));
		
	}).auth(null, null, true, accessToken);
}

and here is the error 

body: {"error":{"code":"7505","message":"Mismatch between request authorization tenant and call tenant.","innerError":{"date":"2021-01-19T12:14:54","request-id":"1f5357a9-65ea-4ffb-a0bf-448cf5577273","client-request-id":"1f5357a9-65ea-4ffb-a0bf-448cf5577273"}}}

**Comment by nomanmuhammad** [comment]
@gclark01 Could you please help me to understand what is my issue. I am calling the bot from skype and receiving the call successfully but not able to answer the call using answer api

**Comment by gclark01** [comment]
@nomanmuhammad - this conversation was about creating calls, looks like you are wanting to answer calls.  Based on the error its seems that there is a mismatch between (maybe caller and app tenants).  I do not believe adding tenantId to the json is going to help here.   I have not tested this scenario personally - if I can find some time later today I will test this out.

**Comment by nomaannomii** [comment]
@gclark01 Thanks for you response. I would love to hear from you if you are able to answer the call.

I am using skype to call this bot but this is not working.

**Comment by lramosvea** [comment]
@ChetanSharma-msft should this issue be closed and keep issue #14635 open?

**Comment by ChetanSharma-msft** [comment]
Hello @Mark-Bailey-SingleWire - I am closing this issue.
Please track it here: https://github.com/microsoftgraph/microsoft-graph-docs/issues/14635
